provider/pkg/gen/schema.go

```diff
@@ -199,12 +202,31 @@ func (g *packageGenerator) findResources(parent string, resources map[string]dis
 			}
 		}
 
+		typeName := fmt.Sprintf("%s%s", parent, name)
+		if override, has := overrides[g.genToken(typeName)]; has {
+			if createMethod == nil {
+				createMethod = override.retrieveCreate(res)
+			}
+			if getMethod == nil {
+				getMethod = override.retrieveGet(res)
+			}
+			if updateMethod == nil {
+				updateMethod = override.retrieveGet(res)
+			}
+			if deleteMethod == nil {
+				deleteMethod = override.retrieveDelete(res)
+			}
```

Comment:
Shall we add a struct with four methods and have a single `retrieve` func to get them all?